storage/include/sparse_merkle/base_types.h

```diff
@@ -95,10 +118,21 @@ class Hash {
   Hash(std::array<uint8_t, SIZE_IN_BYTES> buf) : buf_(buf) {}
 
   bool operator==(const Hash& other) const { return buf_ == other.buf_; }
+  bool operator!=(const Hash& other) const { return buf_ != other.buf_; }
+  bool operator<(const Hash& other) const { return buf_ < other.buf_; }
 
   const uint8_t* data() const { return buf_.data(); }
   size_t size() const { return buf_.size(); }
 
+  // Return buf_ as lowercase hex string
+  std::string toString() const {
+    std::string output;
+    for (size_t i = 0; i < MAX_NIBBLES; i++) {
+      output.push_back(getNibble(i).hexChar());
+    }
+    return output;
+  }
+
```

Comment:
Can we share implementation with `NibblePath::toString()`?

Reply:
Maybe with some template chicanery. But I think it's simple enough that the duplication doesn't really matter here.